### ISSUE TITLE
Invisible mouse cursor

### ISSUE THREAD

**Issue by Devil2548** [Delta2Force/MCVmComputers#14]
Currently using a host pc running windows 10 home(yes I know about the bad performance) with the in-game computer running Ubuntu version 20.04 LTS. Windows version is 1909 build 18363.900 and if it will help here is the 
[latest.log](https://github.com/Delta2Force/MCVmComputers/files/4778121/latest.log)

**Comment by Delta2Force** [comment]
So the cursor is invisble in the VM?

**Comment by Hacker1245** [comment]
Seems this might be an issue with Ubuntu in theVM as the cursor is invisible for me too.

**Comment by Delta2Force** [comment]
If you access the VM directly through VirtualBox, the cursor is visible though right?

**Comment by Hacker1245** [comment]
Yes

**Comment by Devil2548** [comment]
I apologize for not responding as my sleep schedule is currently fairly messed up, but I figured I should respond as well. Hacker1245 describes it just like how it is happening to me. The in-game screen won't show the cursor, but if I access the VM directly through VirtualBox then the cursor shows up just fine.

**Comment by Delta2Force** [comment]
Thats a weird problem. Does the cursor at least flash when it's moved or something?

**Comment by Devil2548** [comment]
No. The cursor remains completely invisible no matter if it is still or moving. The only way to see where the cursor is located is by going off of the fact that certain buttons change in texture when you move the cursor over them(and by right-clicking and seeing where the menu pops up.)

**Comment by Hacker1245** [comment]
> No. The cursor remains completely invisible no matter if it is still or moving. The only way to see where the cursor is located is by going off of the fact that certain buttons change in texture when you move the cursor over them(and by right-clicking and seeing where the menu pops up.)

same behaviour here

**Comment by TheOriginalGolem** [comment]
i'm using the mod v1.4.3, and the am having same problem

**Comment by TheOriginalGolem** [comment]
windows vm is fine but linux vm no cursor, also win7 has no internet

**Comment by Faithy3** [comment]
I just ran into this problem on a Pop!_OS and a Lubuntu VM (windows 10 pro 2004 host, VBox 6.1.12), and after testing with VirtualBox itself, outside of the mod, I think I found the cause, or, at least one: VirtualBox's Mouse Pointer Integration feature. Even when it's disabled, guests that support it will not draw a cursor, despite it being able to interact with everything on the desktop.

**Comment by QQLKAA** [comment]
I think it is because the default VirtualBox mouse mode is a tablet, maybe x11 hides the cursor when it is in tablet mode?

**Comment by Faithy3** [comment]
default mouse mode in VirtualBox is PS/2 Mouse, USB tablet mode shows the same behavior.

**Comment by Meister1593** [comment]
Arch linux, exactly the same problem, no cursor inside Linux Mint ISO in-game. I can click and right click, but cursor is invisible.

**Comment by Ione15** [comment]
Me too Linux mint xfce (latest version + up to date)

**Comment by Meister1593** [comment]
I might have a idea why cursor is not visible, but i will check later.
https://github.com/Delta2Force/MCVmComputers/blob/1caa340b61cf8b4a991d45e8405dcd3215f60fd2/src/main/java/mcvmcomputers/client/gui/GuiFocus.java#L84

**Comment by CoccodrillooXDS** [comment]
@Meister1593 I'm checking that option out

**Comment by CoccodrillooXDS** [comment]
I've checked but it doesn't do anything, if not blocking totally the mouse input 
![immagine](https://user-images.githubusercontent.com/25722501/92410119-618db080-f143-11ea-8c79-e33b964dc021.png)

**Comment by Delta2Force** [comment]
> I might have a idea why cursor is not visible, but i will check later.
> 
> https://github.com/Delta2Force/MCVmComputers/blob/1caa340b61cf8b4a991d45e8405dcd3215f60fd2/src/main/java/mcvmcomputers/client/gui/GuiFocus.java#L84

I don't think that this would cause the VirtualBox cursor to disappear. It only makes the GLFW cursor (your host OS cursor) disappear

**Comment by Devil2548** [comment]
First of all, sorry for kind of just leaving this issue abandoned. Secondly, I think Faithy3 was right when he said that Virtual Box's Mouse Pointer Integration Feature was to blame. I decided to test it by using an OS that did not support Virtual Box's Guest Additions. The OS was Mandrake 9.2(https://archive.org/details/Mandrake92). To my suprise, the cursor showed up just fine.
